pkg/api/handlers/libpod/system.go

```diff
@@ -0,0 +1,73 @@
+package libpod
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/containers/libpod/libpod"
+	"github.com/containers/libpod/pkg/api/handlers/compat"
+	"github.com/containers/libpod/pkg/api/handlers/utils"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/gorilla/schema"
+	"github.com/pkg/errors"
+)
+
+// SystemPrune removes unused data
+func SystemPrune(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+
+	query := struct {
+		All     bool `schema:"all"`
+		Volumes bool `schema:"volumes"`
+	}{}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.Wrapf(err, "Failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+
+	var systemPruneReport = new(entities.SystemPruneReport)
+	fmt.Println("in sytem prune")
+	podPruneReport, err := PodPruneHelper(w, r)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	systemPruneReport.PodPruneReport = podPruneReport
+
+	// We could parallelize this, should we?
+	containerPruneReport, err := compat.PruneContainersHelper(w, r, nil)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	systemPruneReport.ContainerPruneReport = containerPruneReport
+
+	results, err := runtime.ImageRuntime().PruneImages(r.Context(), query.All, nil)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+
+	report := entities.ImagePruneReport{
+		Report: entities.Report{
+			Id:  results,
+			Err: nil,
+		},
+		Size: 0,
+	}
```

Comment:
```suggestion
	report := entities.ImagePruneReport{
		Report: entities.Report{
			Id:  results,
		},
	}
```
Don't initialize zero-value fields.

Reply:
Fixed here. @jwhonce should I fix it here as well? https://github.com/containers/libpod/blob/master/pkg/domain/infra/abi/images.go#L49

Reply:
@sujil02 Please do.

Reply:
@jwhonce done. Please re-review so I can get the ABI and tunnel flow in.